libsolidity/lsp/SemanticTokensBuilder.cpp

```diff
@@ -0,0 +1,301 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#include <libsolidity/lsp/SemanticTokensBuilder.h>
+
+#include <liblangutil/CharStream.h>
+#include <liblangutil/SourceLocation.h>
+
+#include <fmt/format.h>
+
+using namespace std;
+using namespace solidity::langutil;
+using namespace solidity::frontend;
+
+#if 0 // !defined(NDEBUG)
+#define DPRINT(x) do { std::cerr << (x) << std::endl; } while (0)
```

Comment:
This is old take-over code that can go now. Use lspDebug(x) for consistency instead. :)